### PR TITLE
CHE-229: Maven integration

### PR DESCRIPTION
**Pull request by svor**
Features: 
- Maven plugin 
- Classpath configuration
- Effective pom
- Plain java plugin 

https://github.com/eclipse/che-dependencies/pull/6

### CODE REVIEW

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/reconciler/DirtyRegion.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public int getLength() {
     }
 
     /*
-     * @see ITypedRegion#getType
+     * @see ITypedRegion#getProjectType
```
Comment (vparfonov) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/reconciler/DirtyRegion.java]: fix javadoc

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/HasNotificationPanel.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.jseditor.client.texteditor;
+
+import com.google.gwt.dom.client.Element;
+
+/**
+ * @author Evgen Vidolob
```
Comment (skabashnyuk) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/HasNotificationPanel.java:18]: javadoc

File: plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-core-resources/src/main/java/org/eclipse/che/core/internal/resources/Workspace.java [hunk 3]
```diff
@@ -95,29 +95,37 @@
...
+    public Workspace(String path, Provider<ProjectRegistry> projectRegistry, Provider<ProjectManager> projectManager) {
         this.wsPath = path;
-        this.projectManager = projectManager;
         this.projectRegistry = projectRegistry;
+        this.projectManager = projectManager;
+//        try {
+//            projectsRoot = projectManager.getProjectsRoot();
+//        } catch (ServerException e) {
+//            throw new IllegalStateException(e);
+//        }
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-core-resources/src/main/java/org/eclipse/che/core/internal/resources/Workspace.java]: remove if not needed

File: plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-core-resources/src/main/java/org/eclipse/che/core/internal/resources/WorkspaceRoot.java [hunk 4]
```diff
@@ -116,44 +111,40 @@ public IProject getProject(String name) {
...
+//                               RegisteredProject rootProject,
+//                               ProjectManager manager) throws IOException,
+//                                                              ForbiddenException,
+//                                                              ConflictException,
+//                                                              NotFoundException,
+//                                                              ServerException {
+//        List<? extends ProjectConfig> modules = manager.getProjectModules(rootProject);
+//
+//
+//        for (ProjectConfig module : modules) {
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-core-resources/src/main/java/org/eclipse/che/core/internal/resources/WorkspaceRoot.java]: remove if not needed

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/project/classpath/valueproviders/selectnode/SelectNodeViewImpl.ui.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    <ui:style field="style">
+        .main {
+            width: 400px;
+            height: 450px;
+        }
+    </ui:style>
+
+    <g:DockLayoutPanel addStyleNames="{style.main}" ui:field="treeContainer" />
+</ui:UiBinder>
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/project/classpath/valueproviders/selectnode/SelectNodeViewImpl.ui.xml]: new line

File: plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/ClasspathService.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Service for getting information about classpath.
+ *
+ * @author Valeriy Svydenko
+ */
+@Path("jdt/{wsId}/classpath/")
+public class ClasspathService {
+
+    private static final JavaModel model = JavaModelManager.getJavaModelManager().getJavaModel();
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/ClasspathService.java:39]: Why don't you inject this as a constructor parameter using guice provider?

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/MarkDirAsSourceAction.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.java.client.action;
```
Comment (vparfonov) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/MarkDirAsSourceAction.java:11]: use in package name "plugin" instead "ide.ext"

File: plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/ClasspathService.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<ClasspathEntry> getClasspath(@QueryParam("projectpath") String projectPath) throws JavaModelException {
+        IJavaProject javaProject = model.getJavaProject(projectPath);
+        return convertToClasspathEntryDTO(javaProject.getRawClasspath());
+    }
+
+    private List<ClasspathEntry> convertToClasspathEntryDTO(IClasspathEntry[] entries) {
+        if (entries.length == 0) {
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/ClasspathService.java:56]: this private method convertToClasspathEntryDTO is redundant. Please inline.
If ClasspathEntry is DTO it should contains DTO in class name

File: plugins/plugin-java/che-plugin-java-ext-lang-shared/src/main/java/org/eclipse/che/ide/ext/java/shared/dto/classpath/ClasspathEntry.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import org.eclipse.che.dto.shared.DTO;
+
+/**
+ * DTO represents the information about classpath of the project.
+ *
+ * @author Valeriy Svydenko
+ */
+@DTO
+public interface ClasspathEntry {
+    int LIBRARY   = 1;
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-lang-shared/src/main/java/org/eclipse/che/ide/ext/java/shared/dto/classpath/ClasspathEntry.java]: Constants in DTO?

File: plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-ide/src/main/java/org/eclipse/che/plugin/java/plain/client/wizard/PlainJavaProjectWizardRegistrar.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+import static org.eclipse.che.plugin.java.plain.shared.PlainJavaProjectConstants.PLAIN_JAVA_PROJECT_ID;
+
+/**
+ * Provides information for registering Plain Java project type into project wizard.
+ *
+ * @author Valeriy Svydenko
+ */
+public class PlainJavaProjectWizardRegistrar implements ProjectWizardRegistrar {
+    private final List<Provider<? extends WizardPage<ProjectConfigDto>>> wizardPages;
+
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-ide/src/main/java/org/eclipse/che/plugin/java/plain/client/wizard/PlainJavaProjectWizardRegistrar.java]: wizardPages looks useless. Looks like wizardPages  can't  be changed at all

File: plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/rest/ClasspathUpdaterService.java [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+/**
+ * Service for updating classpath.
+ *
+ * @author Valeriy Svydenko
+ */
+@Path("jdt/{wsId}/classpath/update")
+public class ClasspathUpdaterService {
+    private static final JavaModel model = JavaModelManager.getJavaModelManager().getJavaModel();
+
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/rest/ClasspathUpdaterService.java:56]: inject it with provider

File: plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/rest/ClasspathUpdaterService.java [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+     * @throws ConflictException
+     *         if update operation causes conflicts
+     * @throws NotFoundException
+     *         if Project with specified path doesn't exist in workspace
+     * @throws IOException
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public void updateClasspath(@QueryParam("projectpath") String projectPath, List<ClasspathEntry> entries) throws JavaModelException,
+                                                                                                                    ServerException,
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/rest/ClasspathUpdaterService.java]: check please formatting

File: plugins/plugin-maven/che-plugin-maven-ext-client/src/main/java/org/eclipse/che/ide/extension/maven/client/MavenResources.java [hunk 13]
```diff
@@ -25,4 +26,16 @@
...
+
+    interface MavenCss extends CssResource {
+
+        String editorInfoPanel();
+
+        String downloadLink();
+
+        String editorMessage();
+    }
 }
```
Comment (skabashnyuk) [inline plugins/plugin-maven/che-plugin-maven-ext-client/src/main/java/org/eclipse/che/ide/extension/maven/client/MavenResources.java]: new line

File: assembly/assembly-machine-server/pom.xml [hunk 14]
```diff
@@ -23,15 +23,39 @@
     <name>Che Assembly Machine Plugins :: Server</name>
     <dependencies>
         <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
```
Comment (skabashnyuk) [inline assembly/assembly-machine-server/pom.xml:27]: Make sure this dependency will not be included in war classpath

**Pull request merged**